File: notify/v1/notification/create-send-notification-detailed/create-send-notification-detailed.5.x.cs
// Install the C# / .NET helper library from twilio.com/docs/csharp/install

using System;
using System.Collections.Generic;
using Twilio;
using Twilio.Rest.Notify.V1.Service;


class Program 
{
    static void Main(string[] args)
    {
        // Find your Account Sid and Token at twilio.com/console
        const string accountSid = "ACXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX";
        const string authToken = "your_auth_token";

        TwilioClient.Init(accountSid, authToken);

        var apn = new Dictionary<string, Object>()
        {
            {"aps", new Dictionary<string, Object>()
                {
                    {"alert", new Dictionary<string, Object>()
                        {
                            {"title", "Short title for Watch."}
                        }}
                }}
        };

        var data = new Dictionary<string, Object>()
        {
            {"custom_key1", "custom value 1"},
            {"custom_key2", "custom value 2"}
        };

        var fcm = new Dictionary<string, Object>()
        {
            {"notification", new Dictionary<string, Object>()
                {
                    {"title", "New alert"},
                    {"body", "Hello Bob!"}
                }}
        };

        var identity = new List<string> {
            "00000001"
        };

        var notification = NotificationResource.Create(
            apn: apn,
            body: "This is the body for all Bindings",
            data: data,
            fcm: fcm,
            identity: identity,
            title: "Generic loooooooong title for all Bindings",
            pathServiceSid: "ISXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX"
        );

        Console.WriteLine(notification.Sid);
    }
}

File: notify/v1/notification/create-send-notification-detailed/create-send-notification-detailed.5.x.rb
# Download the helper library from https://www.twilio.com/docs/ruby/install
require 'rubygems'
require 'twilio-ruby'

# Your Account Sid and Auth Token from twilio.com/console
account_sid = 'ACXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX'
auth_token = 'your_auth_token'
@client = Twilio::REST::Client.new(account_sid, auth_token)

notification = @client.notify
                      .services('ISXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX')
                      .notifications
                      .create(
                         apn: {
                           'aps' => {
                               'alert' => {
                                   'title' => 'Short title for Watch.'
                               }
                           }
                       },
                         body: 'This is the body for all Bindings',
                         data: {
                           'custom_key1' => 'custom value 1',
                           'custom_key2' => 'custom value 2'
                       },
                         fcm: {
                           'notification' => {
                               'title' => 'New alert',
                               'body' => 'Hello Bob!'
                           }
                       },
                         identity: '00000001',
                         title: 'Generic loooooooong title for all Bindings'
                       )

puts notification.sid
